### PR TITLE
RS/YJ/report module update summary report

### PR DESCRIPTION
**Pull request by yunjoonjung-PNNL**
This PR includes the summary report development.

Although I have a few remaining questions as shown below, I believe the updated code is good enough to submit the PR and is ready to be discussed. The code is fully functional as well.

Note)
1) For some reason, the 6-6 RMD was missing and I re-created/added it here.

2) If the ```result``` key is empty in the ```outcome``` variable (e.g., 5-5, 5-13, etc.), I changed the empty outcome to ```UNDETERMINED```. If this behavior isn't appropriate, please let me know!

3) As Weili and I discussed before and as a reminder, ```90.1-2019 Section``` should be added to the ```report``` variable.

Questions)
1) In the RMD Files section, how would we able to get the user/proposed/baseline file names? This is because when the ```ASHRAE9012019SummaryReport``` class is called from the ```evaluate``` function in ```cli.py```, these file names are NOT passed to the ```RCTReport``` class. Hence, I wonder how we could specify the file names in the ```ashrae901_2019_summary_report.md```.

2) In the ```Summary: All Rules``` table, what section(s) belongs to ```HVAC-General```? I know sections 21 and 22 belong to ```HVAC-WaterSide``` and section 23 belongs to ```HVAC-Airside```. However, I wasn't sure what section would belong to the ```HVAC-General```. Therefore, all the elements in this column are 0 (for now).

3) In the ```Rule Evaluations``` section, What does it mean by ```Number of applicable components```? How could I calculate this?

For questions (1) and (3), I put ``` TO BE UPDATED!!``` string temporarily, but I'll update it as we move along.

### CODE REVIEW

**Review by charliepnnl — APPROVED** [review]
Approved with just a few comments.

File: rct229/cli.py [hunk 1]
```diff
@@ -18,6 +19,7 @@
     "RAW_OUTPUT": EngineRawOutput,
     "RAW_SUMMARY": EngineRawSummary,
     "ASHRAE9012019_DETAIL": ASHRAE9012019DetailReport,
+    "ASHRAE9012019_SUMMARY": ASHRAE9012019SummaryReport,
 }
```
[the remark this answers is not in the document]
Comment (yunjoonjung-PNNL) [reply rct229/cli.py:25]: great point! Fixed.

File: rct229/report_engine/rct_report.py [hunk 2]
```diff
@@ -40,23 +43,26 @@ def generate(self, rct_outcome, report_dir):
 
         """
         invalid_msg = rct_outcome["invalid_rmrs"]
-        outcomes = rct_outcome["outcomes"]
+        outcomes = sorted(
```
[the remark this answers is not in the document]
Comment (yunjoonjung-PNNL) [reply rct229/report_engine/rct_report.py:46]: Fixed.

File: rct229/cli.py [hunk 3]
```diff
@@ -83,7 +85,9 @@ def evaluate(user_rmd, baseline_rmd, proposed_rmd, reports):
     report = evaluate_rmr_triplet(user_rmd, baseline_rmd, proposed_rmd)
     # have report attached.
     for report_type in reports:
-        report_module = REPORT_MODULE[report_type]()
+        report_module = REPORT_MODULE[report_type](
+            user_rmd.name, proposed_rmd.name, baseline_rmd.name
```
[the remark this answers is not in the document]
Comment (yunjoonjung-PNNL) [reply rct229/cli.py:89]: Thanks for letting me know about this Charlie!

File: rct229/report_engine/rct_report.py [hunk 4]
```diff
@@ -19,6 +19,9 @@ def __init__(self):
             RCTOutcomeLabel.UNDETERMINED: 0,
             RCTOutcomeLabel.NOT_APPLICABLE: 0,
         }
+        self.user_rmd = user_rmd
```
[the remark this answers is not in the document]
Comment (yunjoonjung-PNNL) [reply rct229/report_engine/rct_report.py:22]: Fixed.

**Comment by yunjoonjung-PNNL** [comment]
@weilixu 
```Number of applicable components``` was fixed and Charlie's comments were addressed.

Notes)
1) When you run the code, please comment out rules 5-2 and 5-49. 
   i) 5-2's error occurs when the proposed azimuth (0.0) and user azimuth (0.0) are compared. This error is from the ```pint``` module.
   ii) 5-49's error because the core zone doesn't have the ```infiltration``` key, which is correct. I believe this issue will be addressed soon based on our discussion today.
   iii) Only primary rules are included in the report. If this is wrong, please let me know. I'll get it fixed.

Other than these, I believe this PR is good to be merged. I'd appreciate it if you verified one more time.

File: rct229/report_engine/rct_report.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (weilixu) [reply rct229/report_engine/rct_report.py:44]: Great addition, thanks Yun Joon,
One improvement (not urgent) is to adding [natural sort](https://github.com/pnnl/ruleset-checking-tool/blob/develop/rct229/utils/natural_sort.py).
The current implement could sort 11, 12 before 2,3,...
The natural sort can avoid this problem.

File: rct229/report_engine/rct_report.py [hunk 5]
```diff
@@ -40,23 +40,29 @@ def generate(self, rct_outcome, report_dir):
...
-                    RCTOutcomeLabel.FAILED: 0,
-                    RCTOutcomeLabel.UNDETERMINED: 0,
-                    RCTOutcomeLabel.NOT_APPLICABLE: 0,
-                }
-                rule_report = self.generate_rule_report(outcome, rule_outcome_dict)
-                rule_outcome = self.calculate_rule_outcome(rule_outcome_dict)
-                self.add_rule_to_ruleset_report(
-                    ruleset_report, rule_report, rule_outcome
-                )
+                if outcome["primary_rule"]:  # output only primary rules
```
Comment (weilixu) [inline rct229/report_engine/rct_report.py:54]: Good one, thanks for adding this check

I suggest to add a TODO comment here.
The reason is we havent decided whether we want to add secondary rule outcome in the final report or not.

File: rct229/reports/ashrae901_2019_summary_report.py [hunk 6]
```diff
@@ -1,18 +1,190 @@
...
+                RCTOutcomeLabel.UNDETERMINED,
+                RCTOutcomeLabel.NOT_APPLICABLE,
+            ]:
+                self.ruleset_outcome[section_number][result] += rule_outcome_result[
+                    result
+                ]
+                self.ruleset_outcome["All"][result] += rule_outcome_result[result]
+
+        if section_no == "5":
+            _ruleset_outcome_count_helper("Envelope", rule_outcome_result)
```
Comment (weilixu) [inline rct229/reports/ashrae901_2019_summary_report.py:180]: I think in the rule outcome, there is a new key called 'ruleset_section_title'

We can possibly use ruleset_section_title as dictionary to replace the manual string labels here.
In that case, we can remove these if-else statements

**Review by weilixu — APPROVED** [review]
Approved with some comments for improvement

File: rct229/report_engine/rct_report.py (hunk 5 above)
Comment (yunjoonjung-PNNL) [reply rct229/report_engine/rct_report.py:54]: Thanks Weili! I just put a ```TODO``` comment here.

File: rct229/reports/ashrae901_2019_summary_report.py (hunk 6 above)
Comment (yunjoonjung-PNNL) [reply rct229/reports/ashrae901_2019_summary_report.py:180]: Great point! Updated!

**Comment by yunjoonjung-PNNL** [comment]
> Approved with some comments for improvement

In terms of Weili's comment about using the ```natural sort```, I haven't used it (yet) because i) in order to use the ```natural_keys``` function, we need to change the ```rct_outcome["outcomes"]``` format in a way that the ```natural_keys``` function needs (additional step is required) ii) the current code can sort the section numbers in a human order (e.g., '2-3', '2-5', '11-1', '11-4', '12-2', etc.). 

Please let me know if we need any updates in this part. Thanks!

**Pull request merged**